code/modules/projectiles/projectile.dm

```diff
@@ -170,15 +170,29 @@
 			permutated.Add(A)
 		return FALSE
 	else
-		if(A && A.density && !ismob(A) && !(A.flags_1 & ON_BORDER_1)) //if we hit a dense non-border obj or dense turf then we also hit one of the mobs on that tile.
+		if(A && A.density && !ismob(A) && !(A.flags_1 & ON_BORDER_1) && !istype(A,/obj/machinery) && !istype(A,/obj/structure/)) //if we hit a dense non-border obj or dense turf then we also hit one of the mobs on that tile.
 			var/list/mobs_list = list()
+			var/list/machine_list = list()
 			for(var/mob/living/L in target_turf)
 				mobs_list += L
-			if(mobs_list.len)
-				var/mob/living/picked_mob = pick(mobs_list)
-				if(!prehit(picked_mob))
-					return FALSE
-				picked_mob.bullet_act(src, def_zone)
+			for(var/obj/machinery/m in target_turf)
+				if(istype(m,/obj/machinery/power/emitter/energycannon)==FALSE) // Check for energy cannon spam from CTF, shouldn't really need to process them getting hit.

```

Comment:
We won't accept this, try overriding procs in the energycannon itself.

Reply:
Nevermind, I shouldn't need that check.